gen/Sharpliner.SourceGenerator/SharplinerTemplateParametersSourceGenerator.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System;
+using System.CodeDom.Compiler;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Sharpliner.SourceGenerator;
+
+[AttributeUsage(AttributeTargets.Class)]
+public sealed class SharplinerTemplateParametersAttribute : Attribute
```

Comment:
maybe this should be `GenerateTemplateDefinitionParametersAttribute`? as this basically triggers generation of parameter reference for the template definition